From 2712b38be23258a89165908cc1058ce90234b6a2 Mon Sep 17 00:00:00 2001
From: Kuntal Ghosh <kuntal.ghosh@enterprisedb.com>
Date: Thu, 27 Jun 2019 11:24:12 +0530
Subject: [PATCH 09/18] Restrict sorting by offsets

If a transaction operates on multiple offsets one a page, it'll be
difficult to identify whether the undo action for the page have already
been applied if we sort the undo records wrt offsets.
---
 src/backend/access/undo/undoaction.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/backend/access/undo/undoaction.c b/src/backend/access/undo/undoaction.c
index 96766ee..3d83bd8 100644
--- a/src/backend/access/undo/undoaction.c
+++ b/src/backend/access/undo/undoaction.c
@@ -72,10 +72,14 @@ undo_record_comparator(const void *left, const void *right)
 		return -1;
 	else if (luur->uur_block > ruur->uur_block)
 		return 1;
+	/*
+	 * If we're sorting by offset, the logic of skipping already applied
+	 * undo in zheap_undo_actions_page.
 	else if (luur->uur_offset < ruur->uur_offset)
 		return -1;
 	else if (luur->uur_offset > ruur->uur_offset)
 		return 1;
+		*/
 	else if (((UndoRecInfo *) left)->index < ((UndoRecInfo *) right)->index)
 	{
 		/*
-- 
1.8.3.1

